### PR TITLE
MYNEWT-642: Fix Eddystone UID frame format

### PR DESCRIPTION
**Pull request by simonratner**

### CODE REVIEW

File: net/nimble/host/src/ble_eddystone.c [hunk 1]
```diff
@@ -127,12 +130,22 @@ ble_eddystone_set_adv_data_uid(struct ble_hs_adv_fields *adv_fields, void *uid)
...
     svc_data = ble_eddystone_set_svc_data_base(BLE_EDDYSTONE_FRAME_TYPE_UID);
-    memcpy(svc_data, uid, 16);
-    rc = ble_eddystone_set_adv_data_gen(adv_fields, 16);
+
+    rc = ble_hs_hci_util_read_adv_tx_pwr(&tx_pwr);
+    if (rc != 0) {
+        return rc;
+    }
+    svc_data[0] = tx_pwr;
+    memcpy(&svc_data + 1, uid, 16);
```
Comment (sjanc) [inline net/nimble/host/src/ble_eddystone.c]: this should be either  "&svc_data[1]" or "svc_data + 1"

File: net/nimble/host/src/ble_eddystone.c [hunk 2]
```diff
@@ -186,7 +201,7 @@ ble_eddystone_set_adv_data_url(struct ble_hs_adv_fields *adv_fields,
     }
 
     svc_data = ble_eddystone_set_svc_data_base(BLE_EDDYSTONE_FRAME_TYPE_URL);
-    
+
```
Comment (sjanc) [inline net/nimble/host/src/ble_eddystone.c]: unrelated (and violating code style anyway)

Comment (simonratner) [inline net/nimble/host/src/ble_eddystone.c]: Code style requires trailing whitespace?

**Comment by simonratner** [comment]
@ccollins476ad has got these changes already in other commits, this PR can
be closed (and is wrong in a couple of ways anyway).


On 28 Feb. 2017 4:55 am, "Szymon Janc" <notifications@github.com> wrote:

*@sjanc* requested changes on this pull request.

Also please explain in commit why this change is valid (ie by quoting
related spec)
------------------------------

In net/nimble/host/src/ble_eddystone.c
<https://github.com/apache/incubator-mynewt-core/pull/186#discussion_r103446209>
:

>      int rc;

     svc_data = ble_eddystone_set_svc_data_base(BLE_EDDYSTONE_FRAME_TYPE_UID);
-    memcpy(svc_data, uid, 16);
-    rc = ble_eddystone_set_adv_data_gen(adv_fields, 16);
+
+    rc = ble_hs_hci_util_read_adv_tx_pwr(&tx_pwr);
+    if (rc != 0) {
+        return rc;
+    }
+    svc_data[0] = tx_pwr;
+    memcpy(&svc_data + 1, uid, 16);

this should be either "&svc_data[1]" or "svc_data + 1"
------------------------------

In net/nimble/host/src/ble_eddystone.c
<https://github.com/apache/incubator-mynewt-core/pull/186#discussion_r103446299>
:

> @@ -186,7 +201,7 @@ ble_eddystone_set_adv_data_url(struct ble_hs_adv_fields *adv_fields,
     }

     svc_data = ble_eddystone_set_svc_data_base(BLE_EDDYSTONE_FRAME_TYPE_URL);
-
+

unrelated (and violating code style anyway)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/apache/incubator-mynewt-core/pull/186#pullrequestreview-24230890>,
or mute the thread

**Comment by ccollins476ad** [comment]
Simon, would you mind closing this?  I don't seem to have the necessary privileges.

**Pull request closed without merging**